### PR TITLE
[REVIEW] Add pytest-xdist to dev environment.yml

### PR DESCRIPTION
**Pull request by galipremsagar**
Resolves: #6370 

This PR enables the parallel execution of pytests of `cudf`, `dask_cudf` & `custreamz` in CI. The changes also include adding `pytest-xdist` to dev environments.

With these changes, here is the change in pytest execution times in CI:

| module      | without pytest-xdist | with pytest-xdist(n=6)  |
| ----------- | ----------- | -----------|
| cudf      | 1 hr       |  14 min |
| dask_cudf   |    4 min     | 1 min |
| custreamz  |  6 min |  2 min |

Related Integration changes: https://github.com/rapidsai/integration/pull/188

### CODE REVIEW

**Comment by kkraus14** [comment]
@galipremsagar I think we'd need to hook up pytest-xdist into CI in this PR as well: https://github.com/rapidsai/cudf/blob/branch-0.18/ci/gpu/build.sh#L202-L222

**Comment by galipremsagar** [comment]
> @galipremsagar I think we'd need to hook up pytest-xdist into CI in this PR as well: https://github.com/rapidsai/cudf/blob/branch-0.18/ci/gpu/build.sh#L202-L222

Thanks @kkraus14, I was under the impression integration repo was the place pytests were being triggered. Added it here aswell.

File: ci/gpu/build.sh [hunk 1]
```diff
@@ -205,11 +205,11 @@ fi
 
 cd $WORKSPACE/python/cudf
 gpuci_logger "Python py.test for cuDF"
-py.test --cache-clear --basetemp=${WORKSPACE}/cudf-cuda-tmp --junitxml=${WORKSPACE}/junit-cudf.xml -v --cov-config=.coveragerc --cov=cudf --cov-report=xml:${WORKSPACE}/python/cudf/cudf-coverage.xml --cov-report term
+py.test -n 6 --cache-clear --basetemp=${WORKSPACE}/cudf-cuda-tmp --junitxml=${WORKSPACE}/junit-cudf.xml -v --cov-config=.coveragerc --cov=cudf --cov-report=xml:${WORKSPACE}/python/cudf/cudf-coverage.xml --cov-report term
```
Comment (kkraus14) [inline ci/gpu/build.sh:208]: What's the reasoning behind `6` here? Should we set this programmatically based on something?

File: ci/gpu/build.sh [hunk 2]
```diff
@@ -205,11 +205,11 @@ fi
...
 
 cd $WORKSPACE/python/dask_cudf
 gpuci_logger "Python py.test for dask-cudf"
-py.test --cache-clear --basetemp=${WORKSPACE}/dask-cudf-cuda-tmp --junitxml=${WORKSPACE}/junit-dask-cudf.xml -v --cov-config=.coveragerc --cov=dask_cudf --cov-report=xml:${WORKSPACE}/python/dask_cudf/dask-cudf-coverage.xml --cov-report term
+py.test -n 6 --cache-clear --basetemp=${WORKSPACE}/dask-cudf-cuda-tmp --junitxml=${WORKSPACE}/junit-dask-cudf.xml -v --cov-config=.coveragerc --cov=dask_cudf --cov-report=xml:${WORKSPACE}/python/dask_cudf/dask-cudf-coverage.xml --cov-report term
 
 cd $WORKSPACE/python/custreamz
 gpuci_logger "Python py.test for cuStreamz"
```
Comment (kkraus14) [inline ci/gpu/build.sh:215]: Should we use xdist for custreamz tests as well?

**Review by kkraus14 — CHANGES_REQUESTED** [review]

File: ci/gpu/build.sh (hunk 2 above)
Comment (galipremsagar) [reply ci/gpu/build.sh:215]: Sure, I'll need to test it locally first

File: ci/gpu/build.sh (hunk 1 above)
Comment (galipremsagar) [reply ci/gpu/build.sh:208]: CI test jobs have 8vCPUS, but only set `PARALLEL_LEVEL` to 4 to have a stable build. But for pytests we can easily use 6 to 7 cores. That was probably the reason I didn't set it programmatically here.

File: ci/gpu/build.sh (hunk 2 above)
Comment (galipremsagar) [reply ci/gpu/build.sh:215]: Verified locally with multiple levels of parallelism and custreamz pytests are working too.

**Comment by galipremsagar** [comment]
rerun tests

**Comment by galipremsagar** [comment]
rerun tests

File: ci/gpu/build.sh (hunk 1 above)
Comment (kkraus14) [reply ci/gpu/build.sh:208]: @rapidsai/ops any guidance / thoughts here? This looks good to go otherwise.

**Review by ajschmidt8 — APPROVED** [review]

File: ci/gpu/build.sh (hunk 1 above)
Comment (ajschmidt8) [reply ci/gpu/build.sh:208]: looks good to me.

**Review by kkraus14 — APPROVED** [review]

**Pull request merged**